src/base/hip/hip_conversion.cpp

```diff
@@ -637,6 +638,162 @@ namespace rocalution
         return true;
     }
 
+    template <typename ValueType, typename IndexType>
+    bool csr_to_dense_hip(const rocsparse_handle                 sparse_handle,
+                          const rocblas_handle                   blas_handle,
+                          IndexType                              nrow,
+                          IndexType                              ncol,
+                          const MatrixCSR<ValueType, IndexType>& src,
+                          const rocsparse_mat_descr              src_descr,
+                          MatrixDENSE<ValueType>*                dst)
+    {
+        assert(nrow > 0);
+        assert(ncol > 0);
+
+        assert(dst != NULL);
+        assert(sparse_handle != NULL);
+        assert(blas_handle != NULL);
+        assert(src_descr != NULL);
+
+        allocate_hip(nrow * ncol, &dst->val);
+
+        if(DENSE_IND_BASE == 0)
+        {
+            rocsparse_status status = rocsparseTcsr2dense(
+            sparse_handle, nrow, ncol, src_descr, src.val, src.row_offset, src.col, dst->val, nrow);
+            CHECK_ROCSPARSE_ERROR(status, __FILE__, __LINE__);
+        }
+        else
+        {
+            ValueType* temp = NULL;
+            allocate_hip(nrow * ncol, &temp);
+
+            rocsparse_status sparse_status = rocsparseTcsr2dense(
+            sparse_handle, nrow, ncol, src_descr, src.val, src.row_offset, src.col, temp, nrow);
+            CHECK_ROCSPARSE_ERROR(sparse_status, __FILE__, __LINE__);
+
+            ValueType alpha = static_cast<ValueType>(1);
+            ValueType beta = static_cast<ValueType>(0);
+
+            ValueType* B; 
```

Comment:
Doesn't B need to be allocated by the user?

Reply:
I don't actually use B in the routine as I am just transposing the matrix (beta equals zero) but I cant pass NULL in for B in geam

Reply:
Ah right, you are just transposing A

Reply:
i.e. the geam code detects that beta is zero and the corresponding kernel that is called does not use B

Reply:
I should add a comment probably explaining this though

Reply:
Added comment in latest push